### PR TITLE
Add two MRAC siso examples

### PR DESCRIPTION
**Pull request by KybernetikJo**
- direct mrac using mit rule
- direct mrac using lyapunov rule

examples based on Aström & Wittenmark

### CODE REVIEW

**Comment by murrayrm** [comment]
This PR will break once #916 is merged.  The changes needed to use the new I/O system structure ar straightforward (eg, replace `LinearIOSystem` with `StateSpace` or better yet `ss`).  I suggest we hold off on merging this until after #916, but running unit tests now to make sure there are no problems under various configurations.

File: doc/examples.rst
Comment (murrayrm) [inline doc/examples.rst:37]: Perhaps use shorter filenames?  It seems like `mrac_siso_mit` and `mrac_siso_lyapunov` would fit the bill.

File: doc/mrac_siso_direct_lya_rule_statespace.rst
Comment (murrayrm) [inline doc/mrac_siso_direct_lya_rule_statespace.rst:1]: I would change "siso" to "SISO" (since it is an acronym) and capitalize Lyapunov (since it is a proper name).

File: examples/mrac_siso_direct_lya_rule_statespace.py
Comment (murrayrm) [inline examples/mrac_siso_direct_lya_rule_statespace.py:2]: Add author information?

File: examples/mrac_siso_direct_lya_rule_statespace.py
Comment (murrayrm) [inline examples/mrac_siso_direct_lya_rule_statespace.py:32]: This can all be done by a single call to `ct.ss`:
```
io_plant = ct.ss(A, B, C, D, inputs='u', outputs='x', states='x', name='plant')
```

File: examples/mrac_siso_direct_lya_rule_statespace.py
Comment (murrayrm) [inline examples/mrac_siso_direct_lya_rule_statespace.py:48]: As above, use `ss` instead.

File: examples/mrac_siso_direct_lya_rule_statespace.py
Comment (murrayrm) [inline examples/mrac_siso_direct_lya_rule_statespace.py:100]: Use `ct.nlsys` rather than calling the class constructor.

File: examples/mrac_siso_direct_lya_rule_statespace.py
Comment (murrayrm) [inline examples/mrac_siso_direct_lya_rule_statespace.py:112]: Use `ct.interconnect` rather than calling the class constructor.

File: examples/mrac_siso_direct_mit_rule_statespace.py
Comment (murrayrm) [inline examples/mrac_siso_direct_mit_rule_statespace.py:32]: Replace with call to`ct.ss`.

File: examples/mrac_siso_direct_mit_rule_statespace.py
Comment (murrayrm) [inline examples/mrac_siso_direct_mit_rule_statespace.py:48]: Replace with call to`ct.ss`.

File: examples/mrac_siso_direct_mit_rule_statespace.py
Comment (murrayrm) [inline examples/mrac_siso_direct_mit_rule_statespace.py:106]: Replace with call to`ct.nlsys`.

File: examples/mrac_siso_direct_mit_rule_statespace.py
Comment (murrayrm) [inline examples/mrac_siso_direct_mit_rule_statespace.py:118]: Replace with call to`ct.interconnect`.

**Review by murrayrm — CHANGES_REQUESTED** [review]
This PR needs to be rebased on top of the current main branch to catch the changes in #916, which eliminates the `LinearIOSystem` class and adds a `ct.nlsys` factory function.  In addition, it would be good to replace calls to the class constructors (`NonlinearIOSystem`, `InterconnectedSystem`, etc) with calls to factor functions (`ct.nlsys`, `ct.interconnect`, etc).

**Comment by KybernetikJo** [comment]
Thanks for the review. I will incorporate your suggestions.

**Comment by KybernetikJo** [comment]
I need help.

In fact, I'm not sure about the recent commits of this PR. I think I did something wrong. The old files have been merged back into the branch "mrac-examples" by "git rebase main".

In steps 8 of the manual, (https://github.com/python-control/python-control/wiki/How-to-contribute-with-a-pull-request)
which branch must be updated & synchronized, main or feature?

Current situation of "mrac-examples".
- Files and documentation should be ok.
- But many of my commits are suboptimal. They pollute the log history.

Not sure how to fix this.

File: doc/examples.rst
Comment (bnavigator) [inline doc/examples.rst]: This is an artifact of a merge, which must be fixed

Comment (bnavigator) [reply doc/examples.rst]: Oops I was looking at an intermediate commit. Please rebase so that it does not appear in the commit history.

Comment (bnavigator) [reply doc/examples.rst]: In reference to your comment: Do a `git rebase -i main` and "squash" some commits.

Comment (KybernetikJo) [reply doc/examples.rst]: Thx, I will try `git rebase -i main`.

In steps 8 of the manual, (https://github.com/python-control/python-control/wiki/How-to-contribute-with-a-pull-request)
which branch must be updated & synchronized, main or feature?

Comment (bnavigator) [reply doc/examples.rst]: Both. First get your local main branch in sync with upstream and then rebase feature onto your synced local main

**Pull request merged**
